src/core/score_map.cc

```diff
@@ -164,23 +164,21 @@ pair<sds, bool> ReallocIfNeededGeneric(void* obj, float ratio) {
 }  // namespace
 
 bool ScoreMap::iterator::ReallocIfNeeded(float ratio, std::function<void(sds, sds)> cb) {
-  // Unwrap all links to correctly call SetObject()
-  auto* ptr = curr_entry_;
-
-  if (ptr->IsLink()) {
-    ptr = ptr->AsLink();
-  }
-
-  // Note: we do not iterate over the links. Although we could that...
-  auto* obj = ptr->GetObject();
-  auto [new_obj, reallocated] = ReallocIfNeededGeneric(obj, ratio);
-  if (reallocated) {
-    if (cb) {
-      cb((sds)obj, (sds)new_obj);
+  bool reallocated = false;
+  auto body = [ratio, &cb, &reallocated](auto* ptr) {
```

Comment:
nit: since there are many pointers of differrent types, it could be nice to use explicit `DensePtr` here and also below in StringMap

Reply:
I wanted to do the same, the problem is `DensePtr` is private in this context so it can't really be used :frowning_face: Maybe we can make them public ? (I will patch this in a follow up PR with a few boilerplate removed)